lib/ansible/modules/network/asa/asa_og.py

```diff
@@ -0,0 +1,804 @@
+
+    return commands
+
+
+def map_obj_to_commands(want, have, module):
+
+    global have_name
+    have_name = have[0].get('have_name')
+
+    global have_group_type
+    have_group_type = have[0].get('have_group_type')
+
+    global have_config
+    have_config = have[0].get('have_lines')
+
+    global have_description
+    have_description = have[0].get('have_description')
+
+    global have_host_ip
+    have_host_ip = have[0].get('have_host_ip')
+
+    global have_group_object
+    have_group_object = have[0].get('have_group_object')
+
+    global have_ip_mask
+    have_ip_mask = have[0].get('have_ip_mask')
+
+    global have_protocol
+    have_protocol = have[0].get('have_protocol')
+
+    global have_port_range
+    have_port_range = have[0].get('have_port_range')
+
+    global have_port_eq
+    have_port_eq = have[0].get('have_port_eq')
+
+    global have_service_cfg
+    have_service_cfg = have[0].get('have_service_cfg')
+
+    for w in want:
+
+        want_dict = dict()
+
+        want_dict['name'] = w['name']
+        want_dict['group_type'] = w['group_type']
+        want_dict['protocol'] = w['protocol']
+        want_dict['description'] = w['description']
+        want_dict['host_ip'] = w['host_ip']
+        want_dict['group_object'] = w['group_object']
+        want_dict['ip_mask'] = w['ip_mask']
+        want_dict['port_range'] = w['port_range']
+        want_dict['port_eq'] = w['port_eq']
+        want_dict['service_cfg'] = w['service_cfg']
+        state = w['state']
+
+        if state == 'replace':
+            return replace(want_dict)
+        elif state == 'present':
+            return present(want_dict)
+        elif state == 'absent':
+            return absent(want_dict)
+
+
+def map_params_to_obj(module):
+
+    obj = list()
+
+    obj.append({
+        'name': module.params['name'],
+        'group_type': module.params['group_type'],
+        'protocol': module.params['protocol'],
+        'state': module.params['state'],
+        'description': module.params['description'],
+        'host_ip': module.params['host_ip'],
+        'group_object': module.params['group_object'],
+        'port_range': module.params['port_range'],
+        'port_eq': module.params['port_eq'],
+        'service_cfg': module.params['service_cfg'],
+        'ip_mask': module.params['ip_mask']
+    })
+
+    return obj
+
+
+def main():
+
+    argument_spec = dict(
+        name=dict(required=True),
+        group_type=dict(choices=['network-object', 'service-object', 'port-object'], required=True),
+        protocol=dict(choices=['udp', 'tcp', 'tcp-udp']),
+        host_ip=dict(type='list'),
+        description=dict(),
+        group_object=dict(type='list'),
+        ip_mask=dict(type='list'),
+        port_range=dict(type='list'),
+        port_eq=dict(type='list'),
+        service_cfg=dict(type='list'),
+        state=dict(choices=['present', 'absent', 'replace'], default='replace')
+    )
+
+    argument_spec.update(asa_argument_spec)
```

Comment:
This basically adds the provider dict to the module argspec, but remove this.

Reply:
If I remove that I get this error:

`
line 133, in get_config\nKeyError: 'passwords'\n"
`

I believe some `network_cli` arguments have to be passed to `get_config` function.

```
def get_config(module, flags=None):
    flags = [] if flags is None else flags

    passwords = module.params['passwords']
    if passwords:
        cmd = 'more system:running-config'
    else:
        cmd = 'show running-config '
        cmd += ' '.join(flags)
        cmd = cmd.strip()
```

Reply:
@NilashishC Can you please let me know if we can leave that line and close this conversation or if changes are needed. I would love to see this PR merged and pushed for Ver 2.8.
Thanks! :)

Reply:
@FedericoOlivieri I just raised a PR which should mitigate this problem. This is the one https://github.com/ansible/ansible/pull/54783. Once this is merged, please fetch, rebase and try  again.

Reply:
In the meantime, I modified on my dev environment `asa.py` with the changes proposed in #54783 and run a full cycle of integration test and they passed.

Reply:
The fix has been merged. Please fetch, rebase and push the commits. :)